### ISSUE TITLE
pythonPackages.{pygobject3,gst-python} collision on 17.09

### ISSUE THREAD

**Issue by ashgillman** [NixOS/nixpkgs#33932]
## Issue description

collision between `/nix/store/v8j5nk2lhbgs30ld5zgv4p8fsqdicc91-gst-python-1.10.4/lib/python2.7/site-packages/gi/overrides/__init__.py' and `/nix/store/bc3kdjqbw8yvcm80djk1mp20ri26dm6p-python2.7-pygobject-3.24.1/lib/python2.7/site-packages/gi/overrides/__init__.py'

I'm happy to have a crack at fixing. Is there a way to specify a priority for example? If so, which would be preferred.

### Steps to reproduce

nix-build on

```
let pkgs = import <nixpkgs> {};
in
  with pkgs.python27Packages;
  python.buildEnv.override {
    extraLibs = [ gst-python pygobject3 ];
  }
```

Same issue with python35Packages

## Technical details

```
gil at 22:00:55 on nixosx in ~/tmp

♪ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 4.12.10, NixOS, 17.03.1947.d77e0969a78 (Gorilla)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.14`
 - channels(root): `"nixos-17.09.2752.fea24301317"`
 - channels(gil): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`
```

**Comment by jtojnar** [comment]
The file was introduced here https://github.com/NixOS/nixpkgs/commit/36af50f69a1f42d643c3acd8362b3c58db696a95 maybe there is a better solution.

**Comment by jtojnar** [comment]
Also `python.withPackages (pkgs: with pkgs; [ gst-python pygobject3 ])` is usually used.

**Comment by ashgillman** [comment]
Cheers for that last tip.]

Any idea on who knows more about that issue? Committer 宋文武 doesn't seem to have a GitHub account. Neither `gst-python` nor `pygobject` have `meta.maintainers` defined.

**Comment by FRidh** [comment]
The collision appears to be a namespace issue. Typically these only occur on Python 2, because on Python 3 the files would not exist because of PEP 420. In this case, the issue appears to be that the contents of the two files is actually different. Note the issue does not occur on master.

**Comment by ashgillman** [comment]
If not on master, I'm happy for issue to be closed and I'll find a workaround

**Comment by FRidh** [comment]
I'm closing this as `wontfix`, but feel free to fix it on 17.09 :)
